### PR TITLE
doc: fix typos

### PR DESCRIPTION
**Pull request by mika**
s/amount of times/number of times/
s/assosciated/associated/
s/availabe/available/
s/Commiting/Committing/
s/Containter/Container/
s/dependant/dependent/
s/developement/development/
s/filesytem/filesystem/
s/guarenteed/guaranteed/
s/hiearchy/hierarchy/
s/intance/instance/
s/Interger/Integer/
s/mutiple/multiple/
s/nubmer/number/
s/occured/occurred/
s/overriden/overridden/
s/reseted/reset/
s/sytem/system/
s/unkown/unknown/

Signed-off-by: Michael Prokop <mika@grml.org>

## Checklist
- [ ] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug
---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: doc/dev/osd_internals/erasure_coding/developer_notes.rst [hunk 1]
```diff
@@ -207,9 +207,9 @@ set in the erasure code profile, before the pool was created.
   ceph osd erasure-code-profile set myprofile \
      directory=<dir>         \ # mandatory
      plugin=jerasure         \ # mandatory
-     m=10                    \ # optional and plugin dependant
-     k=3                     \ # optional and plugin dependant
-     technique=reed_sol_van  \ # optional and plugin dependant
+     m=10                    \ # optional and plugin dependent
+     k=3                     \ # optional and plugin dependent
+     technique=reed_sol_van  \ # optional and plugin dependent
```
Comment (tchaikov) [inline doc/dev/osd_internals/erasure_coding/developer_notes.rst]: i think "dependant" is not a typo. it's just less commonly used than "dependent". see https://www.merriam-webster.com/dictionary/dependant and https://www.lexico.com/en/definition/dependant

@johnwilkins what do you think?

Comment (mika) [reply doc/dev/osd_internals/erasure_coding/developer_notes.rst]: @tchaikov fair enough, I'm not a native speaker of english, just let me know which version you'd prefer

Comment (tchaikov) [reply doc/dev/osd_internals/erasure_coding/developer_notes.rst]: neither am i =( . let's wait for more inputs before moving forward.

**Comment by tchaikov** [comment]
@jdurgin mind taking a look?

File: doc/dev/osd_internals/erasure_coding/developer_notes.rst (hunk 1 above)
Comment (jdurgin) [reply doc/dev/osd_internals/erasure_coding/developer_notes.rst]: I'd not seen "dependant" before, https://www.merriam-webster.com/dictionary/dependant#note-1 suggests it used to be used in particular cases of British English

**Pull request merged**